spring-cloud-gateway-core/src/main/java/org/springframework/cloud/gateway/route/Route.java

```diff
@@ -177,6 +197,16 @@ public B uri(URI uri) {
 			return getThis();
 		}
 
+		public B metadata(Map<String, Object> options) {
+			this.metadata.putAll(options);
```

Comment:
I would agree with @spencergibb that I would expect this method to do `this.metadata = options` and not add all the `options`.

Reply:
The previous comment was about setter in Route class, but not the RouteBuilder method.
The requested change seems not consistent with the existing method filters
```
public B filters(Collection<GatewayFilter> gatewayFilters) {
			this.gatewayFilters.addAll(gatewayFilters);
			return getThis();
		}
```
and at the same time present replaceFilters method:
```
		public B replaceFilters(List<GatewayFilter> gatewayFilters) {
			this.gatewayFilters = gatewayFilters;
			return getThis();
		}
```
So maybe I should add replaceMetadata builder method instead?

Reply:
@ryanjbaxter what do you think I should replace as you asked or just add replaceMetadata method?

Reply:
I'm going with the same pattern as filters for now.